### PR TITLE
Add `tsfresh` into optional dependencies

### LINKED ISSUE #1245: Add `tsfresh` into optional dependancies
### 🚀 Feature Request

In release `tsfresh==0.20.0` matrixprofile dependency became optional, so now we can try to add `tsfresh` as our optional dependency

### Proposal

1. In `pyproject.toml`
- Add `tsfresh = ~0.20.0` as optional dependency
- Add tsfresh to `classification` extras
- Rebuild `poetry.lock`
2. In github configs:
- Remove installing `poetry run pip install tsfresh==0.19.0` from `tests.yaml` and `notebooks.yaml`
- Add "classification" to extras [here](https://github.com/tinkoff-ai/etna/blob/17a9c04c3555c97011015014c4c331b084769b3c/.github/workflows/test.yml#L178) and [here](https://github.com/tinkoff-ai/etna/blob/17a9c04c3555c97011015014c4c331b084769b3c/.github/workflows/notebooks.yml#L34)
3. In `etna/settings`
- Rewrite the suggestion to install tsfresh to install the "classification" extras, now it should check the existing of deps from "classification" extras
4. In `examples/classification.ipynb` 
- Remove the part about installation tsfresh manully
5. We also can remove tsfresh part from `libs` and import the necessary methods from the original library
- Here we also need to fix the LICENSE text [here](https://github.com/tinkoff-ai/etna/blob/17a9c04c3555c97011015014c4c331b084769b3c/LICENSE#L192)

### Test cases

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look #1245.

## Closing issues
Closes #1245.

### CODE REVIEW

File: etna/experimental/classification/feature_extraction/weasel.py [hunk 1]
```diff
@@ -7,18 +7,25 @@
...
 from scipy.sparse import hstack
 from sklearn.feature_extraction.text import CountVectorizer
 from sklearn.feature_selection import chi2
 from typing_extensions import Literal
 
+from etna import SETTINGS
 from etna.experimental.classification.feature_extraction.base import BaseTimeSeriesFeatureExtractor
 from etna.experimental.classification.utils import padd_single_series
 
+if SETTINGS.classification_required:
```
Comment (Mr-Geekman) [inline etna/experimental/classification/feature_extraction/weasel.py:21]: Are we ok with this?

**Comment by Mr-Geekman** [comment]
I'm not really sure that we want to make `tsfresh` obligatory and replace `etna.libs.tsfresh`. Because it has few dependencies, that we don't really need. For example, `dask[dataframe]`.

File: etna/experimental/classification/feature_extraction/weasel.py (hunk 1 above)
Comment (alex-hse-repository) [reply etna/experimental/classification/feature_extraction/weasel.py:21]: Don't get how it works, if user don't have pyts, what happens when he try to instantiate WEASEL?

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]
Don't we want to refresh the methods in libs with the code from new version of tsfresh if it is updated?

File: etna/experimental/classification/feature_extraction/weasel.py (hunk 1 above)
Comment (Mr-Geekman) [reply etna/experimental/classification/feature_extraction/weasel.py:21]: I reworked importing logic. Probably, it will work better and more clear.

**Review by alex-hse-repository — APPROVED** [review]

**Pull request merged**